From 17daa356813e9aa1f01fbe7176f993306aa7f90b Mon Sep 17 00:00:00 2001
From: Carlos Garnacho <carlosg@gnome.org>
Date: Fri, 6 Nov 2020 17:59:33 +0100
Subject: [PATCH] clutter/event: Do not filter out DEVICE_ADDED/REMOVED events

These devices in x11 are "disabled", that doesn't mean we should refrain
from notifying about them.

Fixes: https://gitlab.gnome.org/GNOME/mutter/-/issues/1476
Fixes: https://gitlab.gnome.org/GNOME/mutter/-/issues/1496

https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/1553

(cherry-picked from commit 34710eabc0dc2154d26296f3121728683af4afe6)
---
 clutter/clutter/clutter-event.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/clutter/clutter/clutter-event.c b/clutter/clutter/clutter-event.c
index 51bec88812..9fa32dfd80 100644
--- a/clutter/clutter/clutter-event.c
+++ b/clutter/clutter/clutter-event.c
@@ -1577,7 +1577,9 @@ _clutter_event_push (const ClutterEvent *event,
   device = clutter_event_get_device (event);
   if (device != NULL)
     {
-      if (!clutter_input_device_get_enabled (device))
+      if (event->type != CLUTTER_DEVICE_ADDED &&
+          event->type != CLUTTER_DEVICE_REMOVED &&
+          !clutter_input_device_get_enabled (device))
         return;
     }
 
-- 
GitLab

